src/resmom/mom_main.c

```diff
@@ -8309,15 +8309,39 @@ main(int argc, char *argv[])
 #if defined(RLIM64_INFINITY)
 	{
 		struct rlimit64 rlimit;
+		int curerror;
 
 		rlimit.rlim_cur = RLIM64_INFINITY;
 		rlimit.rlim_max = RLIM64_INFINITY;
 
 		(void)setrlimit64(RLIMIT_CPU,   &rlimit);
 		(void)setrlimit64(RLIMIT_FSIZE, &rlimit);
 		(void)setrlimit64(RLIMIT_DATA,  &rlimit);
-		(void)getrlimit64(RLIMIT_STACK, &orig_stack_size); /* get for later */
-		(void)setrlimit64(RLIMIT_STACK, &rlimit);
+		if (getrlimit64(RLIMIT_STACK, &orig_stack_size) != -1) {
+			if((orig_stack_size.rlim_cur != RLIM64_INFINITY) && (orig_stack_size.rlim_cur < MIN_STACK_LIMIT)) {
```

Comment:
Can it happen that rlim_cur is between RLIM64_INFINITY and MIN_STACK_LIMIT? do we not want to set rlim_max in that case? Also, if our intent is to set stack limit to MIN_STACK_LIMIT then why don't we set it even when rlim_cur is RLIM64_INFINITY?
I'm still confused about why we are enforcing the rlim_max and rlim_cur both. Should we not be just worried about setting rlim_max to something and leave rlim_cur?

Reply:
Can it happen that rlim_cur is between RLIM64_INFINITY and MIN_STACK_LIMIT?

>*Brem*>> A value of RLIM64_INFINITY indicating no limit.(i:e FULL value; equivalent to 0xffffffffffffffffuLL). Only when process running under modified session by setting limit(ulimit -s unlimited) will be rlim_cur = "RLIM64_INFINITY". 
>By default, most GNU/Linux distribution edicts  8MB. With this conditional expression, we are confirming the **there is limit set for stack size from user system && its value less then "Minimum guarantee stack limit for pbs_mom"**. Only when this condition passes, we wanted to boost the stack limit to expected value 16MB(MIN_STACK_LIMIT). 
As we are setting this stack size value before pbs process become daemon, thus next "fork & exec" will enforce this expected stack size to actual pbs_daemon.

And, this is the condition used across all our PBS Pro daemons. 
 
I'm still confused about why we are enforcing the rlim_max and rlim_cur both. Should we not be just worried about setting rlim_max to something and leave rlim_cur?

>*Brem*>> Its must to set rlim_cur & rlim_max, setrlimit syscall fails if we ignore rlim_cur parameter.

Reply:
Got it, makes to set the rlimit to 16MB when it is less than that. 
Another question - Is it okay to leave stack limit set to RLIM64_INFINITY? (if it is set by default)?
Will having a value of INFINITY affects the child processes or threads that this process spawns?

Reply:
*Good question,*  👍 
By default, right now none of the systems provides INFINITY stack size to protect the OS for following reason.
>Programs that have a legitimate reason to need more stack are rare. if infinite loop happens to contain a recursive function call, the stack would quickly eat all the available memory. 

As you have mentioned, process/threads it create inherits the same stack limit from parent, By saying that, current logic will help admin to set the desired value(>16MB), if any of the JOB needs it. Restricting to the strict size will make the application (PBS Job) will shoot it by itself due to stack crunch.

The corner case of INFINITY stack size happen only if the system admin sets it by __intentional__. Anyway, here admin know what he is doing and expects to set unlimited for all applications including PBS Pro. I feel it __okay__ to adhere the system settings just like other process. My understanding is, we should avoid the case where PBS Pro crashs due to minimum stack size(boosted to 16mb now) and should provide a way for JOB to grow to the _desired_ value.(if admin set more than 16mb, we wont do any change in it). 

I have shared my thoughts, please throw your valuable suggestions on this, I am open to accept new changes. Thanks!

Reply:
Well, I thought having a large stack size for a process affects the stack size of the threads it spawns. In any case, I overlooked the fact that admin can set ulimit to something between 16MB to INFINITY (if they really want to). 
I sign-off on the change. Please squash and rebase